pkg/server/admin.go

```diff
@@ -1397,6 +1397,43 @@ func (s *adminServer) checkReadinessForHealthCheck() error {
 		return status.Errorf(codes.Unavailable, "node is shutting down")
 	}
 
+	if !s.server.sqlServer.acceptingClients.Get() {
```

Comment:
CC @ajwerner. Would just the accepting clients check have been enough to detect the outage from last week caused by the rogue SQL migration (not sure that's actually the right language)?

Reply:
Yes, it would have. We run the startup migrations before this point.